gce-staging-1/k8s.tf

```diff
@@ -0,0 +1,46 @@
+resource "random_id" "gke_username" {
+  byte_length = 32
+}
+
+resource "random_id" "gke_password" {
+  byte_length = 32
+}
+
+resource "google_container_cluster" "gke-staging-1" {
+  name               = "gke-staging-1"
+  zone               = "us-central1-a"
+  initial_node_count = 3
+
+  additional_zones = [
+    "us-central1-b",
+    "us-central1-c",
+    "us-central1-f",
+  ]
+
+  master_auth {
+    username = "${random_id.gke_username.hex}"
+    password = "${random_id.gke_password.hex}"
+  }
+
+  node_config {
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+
+# The following outputs allow authentication and connectivity to the GKE Cluster.
+output "client_certificate" {
+  value = "${google_container_cluster.gke-staging-1.master_auth.0.client_certificate}"
+}
+
+output "client_key" {
+  value = "${google_container_cluster.gke-staging-1.master_auth.0.client_key}"
+}
+
+output "cluster_ca_certificate" {
+  value = "${google_container_cluster.gke-staging-1.master_auth.0.cluster_ca_certificate}"
+}
```

Comment:
Can you add another output named `host` whose value is something similar to `https://10.182.64.100:6443` but with the name of the master instead of the IP?

I put a script at `bin/set-k8s-context` which uses the outputs that you've already included, as well as `host`, and creates a context in your local kubeconfig named after the graph directory.

The `macstadium-staging` and `macstadium-prod-1-cluster` directories also have this rule in their Makefile:

```
.PHONY: context
context:
	$(TOP)/bin/set-k8s-context $(ENV_NAME)
```

So you can run `make context` from a graph directory for a K8s cluster and then be able to connect to that cluster from then on with kubectl.